evm/src/main/java/com/horizen/evm/interop/EvmLog.java

```diff
@@ -47,10 +61,11 @@ public String toString() {
         }
         topicsStr = topicsStr.concat("}");
 
+        String dataString = (data == null) ? "" : Converter.toHexString(data);
```

Comment:
Data (and the other fields, actually) can still be null because they are public non-final fields and so they can always can be changed after construction.